Varient/SignupPage2.js

```diff
@@ -0,0 +1,182 @@
+import React from "react";
+import { Text, View, Button, StyleSheet, TextInput, TouchableOpacity} from "react-native";
+import DateTimePicker from '@react-native-community/datetimepicker';
+
+import AuthContext from "./contexts/AuthContext";
+
+
+class SignupPage2 extends React.Component {
+
+    constructor(props) {
+      super(props);
+      this.state = {
+        firstName:"",
+        lastName:"",
+        isPatient:null,
+        dob:null,
+        gender:"",
+        isDatePickerVisible: false,
+        userInputError: false,        
+        errorMessage: "",
+      };
+    }
+    static contextType = AuthContext;
+
+    
+    getDateDisplay(){
+      if(this.state.dob === null){
+        return null
+      }
+      const month = this.state.dob.getMonth() + 1; //months from 1-12
+      const day = this.state.dob.getUTCDate();
+      const year = this.state.dob.getFullYear();
+
+      return `DOB: ${year}-${month}-${day}`;
+    }
+
+    _inputCheck = () => {
+      const { firstName, lastName, isPatient, dob, gender } = this.state;
+      const requiredField = [firstName, lastName, isPatient, dob, gender];
+    
+      this.setState({ userInputError: true });
+      for (const field of requiredField) {
+        if (field === ""){
```

Comment:
The validity of dob is not checked in here. I think it is because it is initially set to null, and changing it to this solved it on my testing device.
```suggestion
        if (field === "" || !field ){
```